website/docs/concepts/provider_lifecycles.mdx

```diff
@@ -0,0 +1,87 @@
+---
+title: Provider Lifecycles
+---
+
+## When does my Provider get created and disposed?
+
+The states that all different types of providers can go through are the same:
+
+- Uninitialized
+- Creating
+- Alive
+- Disposing
+
+### Uninitialized
+
+An uninitialized provider does not take up any memory since its state is not initialized.
+It will stay that way until an Alive provider or the UI reads or watches it.
+
+### Creating -> Alive
+
+When an uninitialized provider is being read or watched it will be created.
+
+When a provider is created your provider creation function will be run. Any
+providers that you read or watch will be created as needed and their state will be retrieved.
+
+If there are any circular dependencies during this creation process Riverpod will throw an error.
+The best way to fix this error is to redesign your dependencies to have a uni-directional dataflow.
+
+The provider's state is stored in a [ProviderContainer]. In a Flutter app this container is in a [ProviderScope] widget.
+
+
+:::warning
+Do not create a ProviderContainer for use in flutter unless using [UncontrolledProviderScope] and you know what you are doing.
+:::
+
+### Alive
+
+When your provider is Alive, changes to its state will cause dependent providers and/or the dependent UI to rebuild.
+
+### Alive -> Disposing
+
+Providers are not necessarily automatically disposed, and this is where many developers run into issues and questions.
+
+There are two ways for a provider to be disposed.
+
+- When defined using the .autoDispose modifier and no longer being watched by the UI or another provider
+- When the provider is being recreated due to a change in another provider that was watched within the current provider's create function
+  Learn more about this second case by reading [Combining Providers](/docs/concepts/combining_providers)
+
+In both cases you can register actions that will happen before the provider is disposed using `ref.onDispose`.
+A disposed provider is in the uninitialized state again, and will be created on demand when the provider is watched or read again.
```

Comment:
That's only true for autoDispose.

When a provider rebuilds due to ref.watch, it's not passing by the uninitialized state. That's the difference between a FutureProvider going back to  AsyncLoading vs exposing an AsyncData(..., isRefreshing: true)